htsinfer/htsinfer.py

```diff
@@ -37,126 +35,35 @@ class HtsInfer:
-        results: Results container for storing determined library metadata.
     """
     def __init__(
         self,
-        path_1: Path,
-        path_2: Optional[Path] = None,
-        out_dir: Path = Path.cwd() / 'results_htsinfer',
-        tmp_dir: Path = Path(tempfile.gettempdir()) / 'tmp_htsinfer',
-        cleanup_regime: CleanupRegimes = CleanupRegimes.DEFAULT,
-        records: int = 0,
-        threads: int = 1,
-        transcripts_file: Path = (
-            Path(__file__).parent.parent.absolute() /
-            "data/transcripts.fasta.gz"
-        ),
-        read_layout_adapter_file: Path = (
-            Path(__file__).parent.parent.absolute() /
-            "data/adapter_fragments.txt"
-        ),
-        read_layout_min_match_pct: float = 2,
-        read_layout_min_freq_ratio: float = 2,
-        lib_source_min_match_pct: float = 2,
-        lib_source_min_freq_ratio: float = 2,
-        read_orientation_min_mapped_reads: int = 20,
-        read_orientation_min_fraction: float = 0.75,
+        config: Config,
     ):
         """Class constructor."""
-        self.path_1 = path_1
-        self.path_2 = path_2
+        self.config = config
         self.run_id = ''.join(
             choices(string.ascii_uppercase + string.digits, k=5)
         )
-        self.out_dir = Path(out_dir) / self.run_id
-        self.tmp_dir = Path(tmp_dir) / f"tmp_{self.run_id}"
-        self.cleanup_regime = cleanup_regime
-        self.records = records
-        self.threads = threads
-        self.transcripts_file = transcripts_file
-        self.read_layout_adapter_file = read_layout_adapter_file
-        self.read_layout_min_match_pct = read_layout_min_match_pct
-        self.read_layout_min_freq_ratio = read_layout_min_freq_ratio
-        self.lib_source_min_match_pct = lib_source_min_match_pct
-        self.lib_source_min_freq_ratio = lib_source_min_freq_ratio
-        self.read_orientation_min_fraction = read_orientation_min_fraction
-        self.read_orientation_min_mapped_reads = (
-            read_orientation_min_mapped_reads
-        )
-        self.path_1_processed: Path = self.path_1
-        self.path_2_processed: Optional[Path] = self.path_2
-        self.transcripts_file_processed: Path = (
-            self.tmp_dir / self.transcripts_file.stem
-            if self.transcripts_file.suffix == ".gz"
-            else self.tmp_dir / self.transcripts_file.name
+        self.config.args.out_dir = Path(config.args.out_dir) / self.run_id
+        self.config.args.tmp_dir = \
+            Path(config.args.tmp_dir) / f"tmp_{self.run_id}"
+        self.config.args.path_1_processed = self.config.args.path_1
+        self.config.args.path_2_processed = self.config.args.path_2
+        self.config.args.t_file_processed = (
+            config.args.tmp_dir / config.args.transcripts_file.stem
+            if config.args.transcripts_file.suffix == ".gz"
+            else config.args.tmp_dir / config.args.transcripts_file.name
```

Comment:
Ah, this is probably what leads to the `E1101` issues? Seems to me that those should be added to the `Args()` model, maybe as `Optional[Path]` with `None` as default. It's semantically weird, because we are mixing non-CLI args with others, but I don't see a better solution right now (without bigger refactorings)

Reply:
This was very ugly I didn't realize it but removed the defaults and kept Path() instead, unfortunately it is not the place where the warning was coming from, the latest commit has the `# pylint: disable=E1101` comment as suggested in the previous comment.